162 endpoint to send invite to an employee to sign up

itzgeebee opened the pull request:
#### Pre-Submission Checklist

<!-- Go over all points below, and after creating the PR, tick all the checkboxes that apply. -->
<!-- All points should be verified, otherwise, read the CONTRIBUTING guidelines from above-->
<!-- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

-   [x] Your pull request targets the `main` branch.
-   [x] Branch name is descriptive and follows the guidelines [here](https://github.com/andela/bestpractices/wiki/Git-naming-conventions-and-best-practices).
-   [x] All tests pass . Use `git commit --amend` to amend any fixes.

#### Checklist:

<!-- Go over all points below, and after creating the PR, tick the checkboxes that apply. -->
<!-- If you're unsure about any of these, don't hesitate to ask in the Help Contributors room linked above. We're here to help! -->

-   [x] All tests are passing.
-   [x] Tested changes locally.
-   [x] Improved documentation or readme where necessary.
-   [x] Your commit messages format follows our commit message guideline [here](https://github.com/andela/bestpractices/wiki/Git-naming-conventions-and-best-practices#commits).

#### What does this PR do?

This PR adds a new invite-user route 

#### How should this be manually tested?

<!-- Add specific steps on how to test this manually
1. npm run dev
2. test the route /auth/invite-user as an admin

##### Any background context you want to provide?

A route to invite users via email is currently not available, this PR addresses that

#### Does it require any deployments updates ?

1. Add FE_URL to the env variables

Screenshot of mail:
![Screenshot_20220914-180814](https://user-images.githubusercontent.com/61087153/190219760-72bb5e33-80cf-4fbf-8526-5cd308db3a94.jpg)

jocrah commented on src/services/users.js:
```diff
@@ -77,8 +79,40 @@ const signInUserByEmail = async (email, password) => {
+
+	await emailService({ to: email, subject: "Invitation to signup", text })
+
+	const { rows, error } = await db.query(
```
I see this in this particular trend in this file and I'm not sure it works that way.  

Reference here: https://node-postgres.com/api/result

 I know for sure we don't have to restructure the error and throw it ourselves. it will throw on its own

jocrah commented on src/routes/auth.js:
```diff
@@ -33,6 +34,25 @@ router.post(
+		res.status(200).json({
+			status: 'success',
+			data: {
+				inviteInfo
```
kindly transform the data here and also the `inviteInfo` shouldn't be a key

itzgeebee commented:
All fixed @jocrah  Thanks for the feedback.

jocrah commented on src/services/users.js:
```diff
@@ -1,10 +1,12 @@
 const jwt = require("jsonwebtoken")
-const { genPasswordHash, verifyPassword } = require("../lib")
+const { genPasswordHash, verifyPassword, emailLib } = require("../lib")
```
so @itzgeebee looks good but let's start our testing process.  this file keeps growing but we'll like to have a good concept of everything easily so first let's do this:

1. create a `users` folder
2. move this code into it and rename the file to `index.js`
3. then instead of adding the `inviteUser` here. move it to its own file `invite-user.js` and  import it here and export it like we do for the `src/lib/index.js`. just the invite-user though
4. create an `invite-user.test.js` and that's where we will begin the testing

kindly let me know

itzgeebee replied:
Okay boss @jocrah

itzgeebee replied:
am I going to write the test with this issue too @jocrah

jocrah replied:
yea moving forward, the test is now going to be part of the issue, not separate. we're only putting automated checks in place to make sure everything works and if any subsequent updates break the code, we can know as well

jocrah replied:
@itzgeebee

jocrah replied:
actually, its fine for now. i think there's a bit more to do before to make the testing starts so i can approve this without the test.

jocrah commented:
one more thing, @itzgeebee can you kindly add a screenshot of how the email looks like at your end when it is sent?

itzgeebee commented:
> 

Where should I add please? @jocrah . The submission checklist?

jocrah commented:
> Where should I add please? @jocrah . The submission checklist?

@itzgeebee  as a normal comment here or in the issue description if you prefer that.  i think the template usually has Screenshots on it you could use that

itzgeebee commented:
![Screenshot_20220914-180814](https://user-images.githubusercontent.com/61087153/190220092-c3738673-c46a-4a7b-90eb-d0c9dfc4e562.jpg)
@jocrah

jocrah commented on src/services/users.js:
```diff
@@ -77,8 +79,38 @@ const signInUserByEmail = async (email, password) => {
+	await emailLib({ to: email, subject: "Invitation to signup", text })
+
+	const { rows } = await db.query(
+		`INSERT INTO user_invites ("email") VALUES ($1) RETURNING *`, [email])
```
@itzgeebee thinking about this, let's insert before sending the email. if for some reason we send the email and the insert doesn't work the invite link will not work anyway

itzgeebee commented:
@jocrah I have updated it. I also updated something in the isAdmin middleware. I hope you don't mind? It was checking for jobRole instead of role and I changed it to role so it doesn't cause issues.

jocrah commented on src/middleware/isAdmin.js:
```diff
@@ -1,6 +1,6 @@
 
 module.exports = (req, res, next) => {
-	if (req.user.jobRole !== 'admin'){
+	if (req.user.role !== 'admin'){
```
nice, good catch @itzgeebee

jocrah commented on src/services/users.js:
```diff
@@ -5,7 +5,6 @@ const config = require("../config")
 const db = require("../db")
```
there are linting issues in this file @itzgeebee

jocrah commented on src/routes/auth.js:
```diff
@@ -33,6 +34,27 @@ router.post(
 )
 
+router.post(
+	'/invite-user',
```
final thing for now @itzgeebee , let's validate the request like we do for the rest, to ensure we have the right data

itzgeebee replied:
Okay sir

jocrah commented on src/schema.js:
```diff
@@ -192,6 +192,18 @@ const queryArticleTagsSchema = Joi.object({
+const inviteUserSchema = Joi.object({
+	body: {
+		email: Joi.string()
+			.email({
```
lets use just `email()`, the existing ones there are not ideal and they are things we want to change
